frontend/packages/pipelines-plugin/src/components/pipelines/modals/triggers/__tests__/remove-utils.spec.ts

```diff
@@ -0,0 +1,66 @@
+import * as k8s from '@console/internal/module/k8s';
+import { PipelineExampleNames, pipelineTestData } from '../../../../../test-data/pipeline-data';
+import {
+  EventlistenerTestData,
+  EventlistenerTypes,
+} from '../../../../../test-data/event-listener-data';
+import { EventListenerModel, TriggerTemplateModel } from '../../../../../models';
+import { EventListenerKind } from '../../../resource-types';
+import { removeTrigger } from '../remove-utils';
+
+const pipelineData = pipelineTestData[PipelineExampleNames.WORKSPACE_PIPELINE];
+
+describe('removeTrigger', () => {
+  const values = {
+    selectedTrigger: 'pipeline-template',
+  };
+
+  beforeEach(() => {
+    jest.spyOn(k8s, 'k8sKill').mockImplementation((model, data) => Promise.resolve(data));
+    jest.spyOn(k8s, 'k8sPatch').mockImplementation((model, data) => Promise.resolve(data));
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+  // setup mock API calls
+  const setupJestMocks = (eventListener: EventListenerKind) =>
+    jest.spyOn(k8s, 'k8sList').mockImplementation((model) => {
+      if (model.kind === EventListenerModel.kind) {
+        return Promise.resolve([eventListener]);
+      }
+      return Promise.resolve([]);
+    });
+  // assertion helper for the remove trigger resources
+  const expectAndDeleteElResources = (eventListener: EventListenerKind) => {
+    expect(jest.spyOn(k8s, 'k8sKill')).toHaveBeenCalledWith(TriggerTemplateModel, {
+      metadata: {
+        name: values.selectedTrigger,
+        namespace: pipelineData.pipeline.metadata.namespace,
+      },
+    });
+    expect(jest.spyOn(k8s, 'k8sKill')).toHaveBeenCalledWith(EventListenerModel, eventListener);
+  };
+
+  it('expect to remove the TT and EL resources when template.name is used in Eventlistener', async () => {
+    try {
+      setupJestMocks(EventlistenerTestData[EventlistenerTypes.BINDINGS_TEMPLATE_NAME]);
+      await removeTrigger(values, pipelineData.pipeline);
+      expect(jest.spyOn(k8s, 'k8sKill')).toHaveBeenCalledTimes(2);
+      expectAndDeleteElResources(EventlistenerTestData[EventlistenerTypes.BINDINGS_TEMPLATE_NAME]);
+    } catch (e) {
+      fail(e);
+    }
+  });
+
+  it('expect to remove the TT and EL resources when template.ref is used in Eventlistener', async () => {
+    try {
+      setupJestMocks(EventlistenerTestData[EventlistenerTypes.BINDINGS_TEMPLATE_REF]);
+      await removeTrigger(values, pipelineData.pipeline);
+      expect(jest.spyOn(k8s, 'k8sKill')).toHaveBeenCalledTimes(2);
+      expectAndDeleteElResources(EventlistenerTestData[EventlistenerTypes.BINDINGS_TEMPLATE_REF]);
+    } catch (e) {
+      fail(e);
+    }
+  });
+});
```

Comment:
Need another test for when there are multiple TTs tied to the same EventListener. [We don't delete EventListeners if they had more than one TT](https://github.com/openshift/console/blob/master/frontend/packages/pipelines-plugin/src/components/pipelines/modals/triggers/remove-utils.ts#L27-L30).

Also maybe another test on top of that one... for two EventListeners tied to the same TT.

Thanks for writing these tests, should have done them myself when I coded it up 😅